src/Persistent/Implementation/STM.hs

```diff
@@ -24,8 +27,8 @@ import Persistent.Api
 -- FIXME: Remove
 import Test.QuickCheck (generate)
 
-newtype Persist a = Persist (ReaderT (TVar AulaData) IO a)
-  deriving (Functor, Applicative, Monad)
+newtype Persist a = Persist (ExceptT ServantErr (ReaderT (TVar AulaData) IO) a)
```

Comment:
Semantically the order of `ExceptT` and `ReaderT` does not matter. However it might more convenient the other way around.